src/components/inputModal.tsx

```diff
@@ -1,27 +1,162 @@
-import { Modal, Button } from "react-bootstrap";
-import React from "react";
-import { useState } from "react";
+import { Modal, Button, Form } from "react-bootstrap";
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+const schema = yup.object().shape({
+  name: yup.string().required("Name is required"),
+  subject: yup.string().required("Subject is required"),
+  marks: yup.number().positive().integer().min(0).required("Marks are required"),
+  grades: yup.string().required("Grades are required"),
+});
 
 interface ModalProps {
   visible: boolean;
   setVisible: Function;
+  isEdit: boolean;
 }
```

Comment:
It's conventional to prefix interface names with "I". You can use IProps for component props.

Reply:
@mkhfahad00 I think the best one would be 
```suggestion
interface IModalProps {
  visible: boolean;
  setVisible: Function;
  isEdit: boolean;
}
```
This was it would represent by name the it's an interface.